compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -249,14 +249,14 @@ impl<'a> Parser<'a> {
                                     .emit_err(errors::AttrWithoutGenerics { span: attrs[0].span });
                             }
                         }
-                        return Ok((None, TrailingToken::None));
+                        return Ok((None, false));
                     };
 
                     if !this.eat(&token::Comma) {
                         done = true;
                     }
-                    // We just ate the comma, so no need to use `TrailingToken`
-                    Ok((param, TrailingToken::None))
+                    // We just ate the comma, so no need to capture the trailing token.
+                    Ok((param, false))
```

Comment:
Including commas into the `node`'s tokens in `node1, node2, node3` doesn't feel right, but maybe it's ok if proc macro attributes are not supported on this kind of nodes, only `cfg`s.
That's assuming a proc macro should only see `node`'s tokens without comma, and `cfg(FALSE)` should remove tokens including tokens.

On the other hand, proc macros probably *should* see the trailing comma as a part of their input, so they could correctly expand into nothing if necessary, like `cfg(FALSE)`.
That would be a change in behavior, but it should only affect unstable proc macros (?).
That would eliminate the need in `TrailingToken::MaybeComma` entirely, because the commas would always be included and would never be trailing.